File: .ref-Research-V6/usr/source/fort/rt1/csqrt.f
	function csqrt(z1)
	complex csqrt, z1, z2
	dimension dummy(2)
	equivalence (a,z2,dummy(1)), (b,dummy(2))

	z2 = z1
	c = cabs(z2)
	c = sqrt(c)
	b = atan2(b, a)/2.0
	a = c*cos(b)
	b = c*sin(b)
	csqrt = z2
	return
	end

File: .ref-BSD-3/usr/src/libF77/c_sqrt.c
#include "complex"

c_sqrt(r, z)
complex *r, *z;
{
double mag, sqrt(), cabs();

if( (mag = cabs(z->real, z->imag)) == 0.)
	r->real = r->imag = 0.;
else if(z->real > 0)
	{
	r->real = sqrt(0.5 * (mag + z->real) );
	r->imag = z->imag / r->real / 2;
	}
else
	{
	r->imag = sqrt(0.5 * (mag - z->real) );
	if(z->imag < 0)
		r->imag = - r->imag;
	r->real = z->imag / r->imag /2;
	}
}
